### PR TITLE
Update to SpeziScheduler 1.0

### PR DESCRIPTION
**Pull request by Supereg**
# Update to SpeziScheduler 1.0

## :recycle: Current situation & Problem
This PR updates the SpeziTemplateApplication to use the new SpeziScheduler 1.0 release. This allow use to remove a lot of code that was previously necessary in the template application. Generally, we were able to greatly simplify the Scheduler setup.

## :gear: Release Notes 
* Upgrade SpeziScheduler to 1.0

## :books: Documentation
We updated the documentation to reflect the changes done to the template app.

## :white_check_mark: Testing
Existing UI test cases are reused.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: TemplateApplication.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1096,24 +1086,24 @@
...
 			};
 		};
 		2F49B7742980407B00BCB272 /* XCRemoteSwiftPackageReference "Spezi" */ = {
 			isa = XCRemoteSwiftPackageReference;
-			repositoryURL = "https://github.com/StanfordSpezi/Spezi";
+			repositoryURL = "https://github.com/StanfordSpezi/Spezi.git";
 			requirement = {
-				kind = upToNextMajorVersion;
-				minimumVersion = 1.7.1;
+				branch = "feature/application-for-swiftui";
```
Comment (PSchmiedmayer) [inline TemplateApplication.xcodeproj/project.pbxproj:1097]: TODO to update this to an release version.

File: TemplateApplication.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1128,8 +1118,8 @@
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/StanfordSpezi/SpeziAccount.git";
 			requirement = {
-				kind = upToNextMajorVersion;
-				minimumVersion = "2.0.0-beta.7";
+				branch = "feature/swift-syntax-6";
```
Comment (PSchmiedmayer) [inline TemplateApplication.xcodeproj/project.pbxproj:1121]: TODO to update this to an release version.

File: TemplateApplication.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1236,6 +1226,22 @@
...
+			requirement = {
+				kind = upToNextMajorVersion;
+				minimumVersion = 2.0.2;
+			};
+		};
+		A94DDFFB2CBD1190004930BD /* XCRemoteSwiftPackageReference "SpeziNotifications" */ = {
+			isa = XCRemoteSwiftPackageReference;
+			repositoryURL = "https://github.com/StanfordSpezi/SpeziNotifications";
+			requirement = {
+				branch = "feature/initial-version";
```
Comment (PSchmiedmayer) [inline TemplateApplication.xcodeproj/project.pbxproj:1241]: TODO to update this to an release version.

File: TemplateApplication/Home.swift [hunk 4]
```diff
@@ -23,16 +23,12 @@ struct HomeView: View {
...
-                .tag(Tabs.schedule)
-                .tabItem {
-                    Label("SCHEDULE_TAB_TITLE", systemImage: "list.clipboard")
-                }
-            Contacts(presentingAccount: $presentingAccount)
-                .tag(Tabs.contact)
-                .tabItem {
-                    Label("CONTACTS_TAB_TITLE", systemImage: "person.fill")
-                }
+            Tab("SCHEDULE_TAB_TITLE", systemImage: "list.clipboard", value: .schedule) {
```
Comment (PSchmiedmayer) [inline TemplateApplication/Home.swift:26]: Nice!

File: TemplateApplication/Schedule/ScheduleView.swift [hunk 5]
```diff
@@ -10,54 +10,53 @@ import OrderedCollections
...
-                    ForEach(eventContextsByDate[startOfDay] ?? [], id: \.event) { eventContext in
-                        EventContextView(eventContext: eventContext)
-                            .onTapGesture {
-                                if !eventContext.event.complete {
-                                    presentedContext = eventContext
-                                }
-                            }
-                    }
+            TodayList { event in
+                InstructionsTile(event) { // TODO: use TASK_CONTEXT_ACTION_QUESTIONNAIRE for the footer Action button!
```
Comment (PSchmiedmayer) [inline TemplateApplication/Schedule/ScheduleView.swift:54]: Very cool to see the simplification in here!

Pointing out the TODO remaining here.

File: TemplateApplication/Schedule/TemplateApplicationScheduler.swift [hunk 6]
```diff
@@ -7,43 +7,41 @@
...
+            try scheduler.createOrUpdateTask(
+                id: "social-support-questionnaire",
+                title: "TASK_SOCIAL_SUPPORT_QUESTIONNAIRE_TITLE",
+                instructions: "TASK_SOCIAL_SUPPORT_QUESTIONNAIRE_DESCRIPTION",
+                schedule: .daily(hour: 8, minute: 0, startingAt: .today)
+            ) { context in
+                context.questionnaire = Bundle.main.questionnaire(withName: "SocialSupportQuestionnaire")
+            }
+        } catch {
+            // TODO: we should visualize this? or at least allow to visualize it!
```
Comment (PSchmiedmayer) [inline TemplateApplication/Schedule/TemplateApplicationScheduler.swift:34]: Maybe a view state that is exposed to the UI and could be used to tigger a view site alert?

File: TemplateApplication/Schedule/TemplateApplicationScheduler.swift [hunk 7]
```diff
@@ -7,43 +7,41 @@
...
-        self.init(tasks: [Self.socialSupportTask])
-    }
+
+extension Task.Context {
+    @Property var questionnaire: Questionnaire?
+}
+
+
+extension Outcome {
+    @Property var questionnaireResponse: QuestionnaireResponse?
```
Comment (PSchmiedmayer) [inline TemplateApplication/Schedule/TemplateApplicationScheduler.swift:46]: Such a cool mechanism!

File: TemplateApplication/Schedule/TemplateApplicationScheduler.swift (hunk 6 above)
Comment (Supereg) [reply TemplateApplication/Schedule/TemplateApplicationScheduler.swift]: We added a viewState that is shown as an alert.

File: TemplateApplication/Schedule/ScheduleView.swift (hunk 5 above)
Comment (Supereg) [reply TemplateApplication/Schedule/ScheduleView.swift]: Resolved 🚀

File: TemplateApplication.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -1118,8 +1117,8 @@
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/StanfordSpezi/SpeziAccount.git";
 			requirement = {
-				branch = "feature/swift-syntax-6";
-				kind = branch;
+				kind = revision;
+				revision = 7ef29a7743c01a083ba69e93479e82d3be199ac0;
```
Comment (PSchmiedmayer) [inline TemplateApplication.xcodeproj/project.pbxproj:1121]: Might make sense to wait to tag a full version here?

File: TemplateApplication.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 9]
```diff
@@ -276,8 +275,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/StanfordSpezi/SpeziScheduler.git",
       "state" : {
-        "branch" : "feature/il-notifications-support",
-        "revision" : "7ad6b40b39bf85536dcbd1b9ac8e8f8f3f86dbbd"
+        "branch" : "main",
+        "revision" : "7ba1ed88760f27cc6d80f75d2993a59f0823be04"
```
Comment (PSchmiedmayer) [inline TemplateApplication.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:279]: Would be great if we tag use a tagged version here.

File: TemplateApplication.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 10]
```diff
@@ -186,17 +186,16 @@
...
       }
     },
     {
       "identity" : "speziaccount",
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/StanfordSpezi/SpeziAccount.git",
       "state" : {
-        "branch" : "feature/swift-syntax-6",
-        "revision" : "d2bfede6e4ac07e4dd5dc2d3dc3fa564de9226a6"
+        "revision" : "7ef29a7743c01a083ba69e93479e82d3be199ac0"
```
Comment (PSchmiedmayer) [inline TemplateApplication.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:198]: Would be great if we tag use a tagged version here.

File: TemplateApplication/Account/AccountSetupHeader.swift [hunk 11]
```diff
@@ -17,7 +17,7 @@ struct AccountSetupHeader: View {
     
     var body: some View {
         VStack {
-            Text("ACCOUNT_TITLE")
+            Text("Your Account")
```
Comment (PSchmiedmayer) [inline TemplateApplication/Account/AccountSetupHeader.swift:20]: Nice, thank you for improving this!

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for working on this; happy to see this merged once we move to fully tagged version in the template app; we should also move Foundation, Accounts, and others to the tagged versions that are not betas 👍

**Comment by Supereg** [comment]
@PSchmiedmayer it seems like periphery doesn’t support swift 6? Or do you have a clue what the issue could be with the failing build?

**Comment by PSchmiedmayer** [comment]
Seem like this might be resolved soon: https://github.com/peripheryapp/periphery/issues/813

Wondering we should temporarily remove CodeQL and Periphery and add an issue to re-add them once they support Xcode 16/Swift 6?

Or maybe make them non-required to merge a PR?

**Comment by PSchmiedmayer** [comment]
Good point; I think we can make both optional for now but keep the actions in there.

I have asked for an estimate on the CodeQL Swift 6 adoption in their GitHub repo: https://github.com/github/codeql-action/issues/2506.
There seems to be some ongoing work, I am sure they are working on supporting this in the near future: https://github.com/github/codeql/pull/17699

**Comment by Supereg** [comment]
> Good point; I think we can make both optional for now but keep the actions in there.
> 
> I have asked for an estimate on the CodeQL Swift 6 adoption in their GitHub repo: [github/codeql-action#2506](https://github.com/github/codeql-action/issues/2506). There seems to be some ongoing work, I am sure they are working on supporting this in the near future: [github/codeql#17699](https://github.com/github/codeql/pull/17699)
> 
> It seems like some of the UI tests are still failing at this point; I would be fine with going ahead and merging the PR with CodeQL and Periphery failing for now (they will work again once support is added to these tools).

Just found, that Periphery only fails because of the project format being Xcode 16.0. I just reverted to the Xcode 15.3 project format as it isn't really important and periphery works again. Now we are just waiting for CodeQL.

Another thing I noticed was that the Build and Test currently always choses the iPhone SE which makes some of our unit test fail as we would need to add additional behavior for scrolling. I tried to change the device to iPhone 16 Pro in Fastlane, however that didn't do anything. Do you know what the issue might be?

**Comment by PSchmiedmayer** [comment]
Very cool; that's great regarding Periphery!

Strange thing regarding the simulator; must be related to what simulators are installed on the runners. I thought all of them should be installed, but there might be a mismatch ...
I will take a look at this 👍

**Comment by PSchmiedmayer** [comment]
@Supereg I updated some of our agents and the builds seem to run now; just seems to be a UI testing issue with our license test that would stop us from merging the PR 👍

**Comment by Supereg** [comment]
> @Supereg I updated some of our agents and the builds seem to run now; just seems to be a UI testing issue with our license test that would stop us from merging the PR 👍

Great news. I’ll make sure to resolve that 👍

**Comment by PSchmiedmayer** [comment]
Thank you 🎉

**Comment by Supereg** [comment]
> Thank you 🎉

Apparently, this was just a Springboard Reboot in the Simulator while executing this test case. Everything passes now.

**Pull request merged**